src/main/java/edu/kit/hci/soli/controller/BookingsController.java

```diff
@@ -39,6 +39,35 @@ public String userBookings(Model model, HttpServletResponse response, Principal
         return "bookings";
     }
 
+    // This is a bit hacky... but it works
+    @GetMapping("/bookings/delete/{id}")
+    public String deleteBookings(@PathVariable("id") Long id, Model model, HttpServletResponse response, Principal principal) {
+        log.info("Received delete request for booking {}", id);
+        User user = userService.resolveLoggedInUser(principal);
```

Comment:
Have you ensured this actually works? I recall an issue leading to this method returning null.